### ISSUE TITLE
test-suite fails with fatal: making test-suite.log: failed to create imfile-escapelf.replacement.trs

### ISSUE THREAD

**Issue by mbiebl** [rsyslog/rsyslog#4146]
Version: 8.2001.0

Running the test-suite of 8.2001.0 (with a fix for #4120 cherry-picked) I run into this `make check` failure:

```
make[6]: Verzeichnis „/home/michael/debian/build-area/rsyslog-8.2001.0/tests“ wird betreten
make[6]: Für das Ziel „imfile-escapelf.replacement.log“ ist nichts zu tun.
make[6]: Für das Ziel „imfile-escapelf.replacement-empty.log“ ist nichts zu tun.
make[6]: Verzeichnis „/home/michael/debian/build-area/rsyslog-8.2001.0/tests“ wird verlassen
fatal: making test-suite.log: failed to create imfile-escapelf.replacement.trs
fatal: making test-suite.log: failed to create imfile-escapelf.replacement.log
fatal: making test-suite.log: failed to create imfile-escapelf.replacement-empty.trs
fatal: making test-suite.log: failed to create imfile-escapelf.replacement-empty.log
make[5]: *** [Makefile:3414: test-suite.log] Fehler 1
make[5]: Verzeichnis „/home/michael/debian/build-area/rsyslog-8.2001.0/tests“ wird verlassen
make[4]: *** [Makefile:3523: check-TESTS] Fehler 2
make[4]: Verzeichnis „/home/michael/debian/build-area/rsyslog-8.2001.0/tests“ wird verlassen
make[3]: *** [Makefile:3590: check-am] Fehler 2
make[3]: Verzeichnis „/home/michael/debian/build-area/rsyslog-8.2001.0/tests“ wird verlassen
make[2]: *** [Makefile:919: check-recursive] Fehler 1
make[2]: Verzeichnis „/home/michael/debian/build-area/rsyslog-8.2001.0“ wird verlassen
dh_auto_test: error: make -j1 check VERBOSE=1 returned exit code 2
```

**Comment by mbiebl** [comment]
Ah, seems those files are simply missing from the dist tarball
```
$ ls test/imfile-escapelf.replacement-empty.sh tests/imfile-escapelf.replacement.sh
ls: cannot access 'test/imfile-escapelf.replacement-empty.sh': No such file or directory
ls: cannot access 'tests/imfile-escapelf.replacement.sh': No such file or directory
```

**Comment by mbiebl** [comment]
Those test files need to be added to EXTRA_DIST unconditionally.
It appears that the dist tarball was built without ENABLE_IMFILE_TESTS being set.

**Comment by mbiebl** [comment]
To avoid duplication, maybe it would make sense to introduce variables like
`IMFILE_TEST_FILES = imfile-basic.sh ...`
And then you have
```
if ENABLE_IMFILE_TESTS
TESTS += $(IMFILE_TEST_FILES)
endif

...
EXTRA_DIST = $(IMFILE_TEST_FILES)
```
I guess you get the idea

**Comment by mbiebl** [comment]
Fixed in 7c2eaa8f2e3b60a12f0c728b6fb11419dc8a4b2a
